src/kmergraph.cpp

```diff
@@ -238,20 +218,271 @@ void KmerGraph::discover_k() {
     }
 }
 
-void KmerGraph::set_exp_depth_covg(const uint32_t edp) {
+std::ostream &operator<<(std::ostream &out, KmerGraph const &data) {
+    for (const auto &c: data.nodes) {
+        out << *(c) << std::endl;
+    }
+    return out;
+}
+
+//save the KmerGraph as gfa
+void KmerGraph::save(const std::string &filepath, const std::shared_ptr<LocalPRG> localprg) {
+    uint32_t sample_id = 0;
+
+    std::ofstream handle;
+    handle.open(filepath);
+    if (handle.is_open()) {
+        handle << "H\tVN:Z:1.0\tbn:Z:--linear --singlearr" << std::endl;
+        for (const auto &c : nodes) {
+            handle << "S\t" << c->id << "\t";
+
+            if (localprg != nullptr) {
+                handle << localprg->string_along_path(c->path);
+            } else {
+                handle << c->path;
+            }
+
+            //TODO: leave as coverage 0 or change this?
+            handle << "\tFC:i:" << 0 << "\t" << "\tRC:i:" << 0 << std::endl;//"\t" << (unsigned)nodes[i].second->num_AT << endl;
+
+            for (uint32_t j = 0; j < c->outNodes.size(); ++j) {
+                handle << "L\t" << c->id << "\t+\t" << c->outNodes[j].lock()->id << "\t+\t0M" << std::endl;
+            }
+        }
+        handle.close();
+    } else {
+        std::cerr << "Unable to open kmergraph file " << filepath << std::endl;
+        std::exit(EXIT_FAILURE);
+    }
+}
+
+void KmerGraph::load(const std::string &filepath) {
+    clear();
+    uint32_t sample_id = 0;
+
+    std::string line;
+    std::vector<std::string> split_line;
+    std::stringstream ss;
+    uint32_t id = 0, covg, from, to;
+    prg::Path p;
+    uint32_t num_nodes = 0;
+
+    std::ifstream myfile(filepath);
+    if (myfile.is_open()) {
+
+        while (getline(myfile, line).good()) {
+            if (line[0] == 'S') {
+                split_line = split(line, "\t");
+                assert(split_line.size() >= 4);
+                id = std::stoi(split_line[1]);
+                num_nodes = std::max(num_nodes, id);
+            }
+        }
+        myfile.clear();
+        myfile.seekg(0, myfile.beg);
+        nodes.reserve(num_nodes);
+        std::vector<uint16_t> outnode_counts(num_nodes + 1, 0), innode_counts(num_nodes + 1, 0);
+
+        while (getline(myfile, line).good()) {
+            if (line[0] == 'S') {
+                split_line = split(line, "\t");
+                assert(split_line.size() >= 4);
+                id = stoi(split_line[1]);
```

Comment:
`id` of what? Suggest a more descriptive name

Reply:
I did not look into details of the `KmerGraph::load` function. This part of the code seems "new" because it is in green, but it was just duplicated into the `KmerGraphWithCoverage`. I will fix this duplication soon, but I think @rmcolq is the fittest to know which descriptive name would be the best.